unsafe/src/main/java/org/apache/spark/unsafe/memory/TaskMemoryManager.java

```diff
@@ -175,12 +175,8 @@ public void free(MemoryBlock memory) {
    * This address will remain valid as long as the corresponding page has not been freed.
    */
   public long encodePageNumberAndOffset(MemoryBlock page, long offsetInPage) {
-    if (inHeap) {
-      assert (page.pageNumber != -1) : "encodePageNumberAndOffset called with invalid page";
-      return (((long) page.pageNumber) << 51) | (offsetInPage & MASK_LONG_LOWER_51_BITS);
-    } else {
-      return offsetInPage;
-    }
+    assert (page.pageNumber != -1) : "encodePageNumberAndOffset called with invalid page";
+    return (((long) page.pageNumber) << 51) | (offsetInPage & MASK_LONG_LOWER_51_BITS);
```

Comment:
i think everywhere we use a magic number like 51, we should have a very brief comment around it to say what that is